features/step_definitions/view_restaurant_page_steps.rb

```diff
@@ -1,6 +1,11 @@
 Given("the following restaurants exist") do |table|
+  x = 0
+  @res_category = RestaurantCategory.all
   table.hashes.each do |hash|
     FactoryGirl.create(:restaurant, hash)
+    restaurant = Restaurant.all[x]
+    @res_category.all[x].update(restaurants: [restaurant])
+    x += 1
```

Comment:
With this solution will each restaurant belong to every category? Feels weird to me

Reply:
with this solution it assigned the restaurants to categories according to table order, it has the drawback of having same amount of categories as restaurants, I am working on using the table to assign restaurant to  categories,

Reply:
I have managed to use the table to assign restaurants to categories but its still showing nil in association field when it gets to the view.